azure/scope/managedmachinepool.go

```diff
@@ -155,6 +155,11 @@ func buildAgentPoolSpec(managedControlPlane *infrav1exp.AzureManagedControlPlane
 		replicas = *machinePool.Spec.Replicas
 	}
 
+	subnetName := managedControlPlane.Spec.VirtualNetwork.Subnet.Name
```

Comment:
the main difference here with AzureCluster subnets is that AzureCluster network spec contains _all_ the cluster subnets, and each AzureMachinePool/AzureMachine subnet must be defined in the AzureCluster too, so that the subnets can be reconciled, how do we plan on dealing with this for managed clusters?

Reply:
@CecileRobertMichon why do we need to manage/reconcile subnets separately ?